test/models/user_test.rb

```diff
@@ -335,4 +335,194 @@ def test_notes_template_validation
     assert(u.invalid?,
            "Notes template with duplication headings should be invalid")
   end
+
+  def test_disable_account
+    rolf.disable_account
+    rolf.reload
+    assert_blank(rolf.password)
+    assert_blank(rolf.email)
+    assert_blank(rolf.mailing_address)
+    assert_blank(rolf.notes)
+    assert_true(rolf.blocked)
+  end
+
+  def test_delete_api_keys
+    assert_not_zero(ApiKey.where(user: rolf).count)
+    rolf.delete_api_keys
+    assert_zero(ApiKey.where(user: rolf).count)
+  end
+
+  def test_delete_interests
+    assert_not_zero(Interest.where(user: junk).count)
+    junk.delete_interests
+    assert_zero(Interest.where(user: junk).count)
+  end
+
+  def test_delete_notifications
+    assert_not_zero(Notification.where(user: rolf).count)
+    rolf.delete_notifications
```

Comment:
Once I get past the prior line, this blows up with:
```ruby
ERROR UserTest#test_delete_notifications (334.58s)
Minitest::UnexpectedError:         ActiveRecord::NotNullViolation: Mysql2::Error: Column 'user_id' cannot be null
            app/models/user.rb:991:in `delete_notifications'
            test/models/user_test.rb:364:in `test_delete_notifications'
```
I'm way out of my depth here.